File: src/python/PythonSearchEngine/search_from_file.py
from search_engine import *
from file_handling import *
import string
import csv

def search_from_file(csv_file):

    query = input('Enter the query here: ')
    tokenized_query = remove_puncts(query, string).split()

    corpus = []
    data = {}

    with open(csv_file, 'r', encoding='utf-8') as csvf:
        csvReader = csv.DictReader(csvf)

        key = 0
        for rows in csvReader:
            data[key] = rows
            key += 1

    for i in range(len(data)):
        corpus.append(' '.join(data[i].values()))

    tokenized_corpus = [remove_puncts(doc, string).split() for doc in corpus]

    bm25 = BM25Okapi(tokenized_corpus)

    scores = bm25.get_scores(tokenized_query)

    cnt = 0
    for score in scores:
        if score != 0.0:
            cnt += 1

    lines1 = bm25.get_top_n(tokenized_query, corpus, n=cnt)

    with open('../resource/answer.txt', 'w') as f:
        for line in lines1:
            f.write(line + "\n")

File: src/main/java/javasearchengine/searchengine/SearchFromFile.java
package javasearchengine.searchengine;

import java.util.*;
import java.io.*;

public class SearchFromFile {

    public static List<String> searchFromFile(String csvFile, String query) throws FileNotFoundException, IOException {
        String[] list = FileHandling.removePuncts(query).split("\\s+");
        List<String> tokenizedQuery = new ArrayList<>();
        for (String x : list) {
            tokenizedQuery.add(x);
        }

        List<String> corpus = new ArrayList<>();
        List<String> columns = new ArrayList<>();

        BufferedReader csvReader = new BufferedReader(new FileReader(csvFile));
        String row;
        boolean check = false;
        while ((row = csvReader.readLine()) != null) {
            if (check == false) {
                for (String x : row.split(",")) {
                    columns.add(x);

                }
                check = true;
            } else {
                row = row.replaceAll(",", " ");
                row = row.replaceAll("\"", "");
                corpus.add(row);
            }
        }
        csvReader.close();

        List<List<String>> tokenizedCorpus = new ArrayList<>();
        for (String doc : corpus) {
            String[] x = FileHandling.removePuncts(doc).split("\\s+");
            List<String> y = new ArrayList<>();
            for (String word : x) {
                y.add(word);
            }
            tokenizedCorpus.add(y);
        }

        BM25Okapi bm25 = new BM25Okapi(tokenizedCorpus);

        List<Double> scores = bm25.getScores(tokenizedQuery);

        int cnt = 0;
        for (double score : scores) {
            if (score != 0.0) {
                cnt++;
            }
        }

        List<String> lines1 = bm25.getTopN(tokenizedQuery, corpus, 10);

        return lines1;

    }

}

